journeyapps/powersync-supabase-flutter-todolist-demo
# feat: add full text search example

## Description
Add Full-Text Search example using Search Bar

## Work Done
* Added search functionality to the app bar
* Added search suggestions generated from the FTS5 table
* Clicking on a search suggestion takes you to the page

## How to test
**Ideally you should have no lists to start because the FTS table will not be populated with already generated lists**
* Run the app
* The table and triggers will be created
* Add lists (e.g "test", "i like to test", "test best friends")
* Click on the search icon at the top bar
* Search (e.g. "test)
* Click on a search result
* You should be directed to the todo list item

## Screenshots
**Create lists**
<img width="418" alt="image" src="https://github.com/powersync-ja/powersync-supabase-flutter-todolist-demo/assets/46312751/789977a8-f840-436e-82af-eea4b16643bd">

**Search**
<img width="415" alt="image" src="https://github.com/powersync-ja/powersync-supabase-flutter-todolist-demo/assets/46312751/8ad6453b-316b-4a77-910b-f2f11cce38db">

**Click on search result**
<img width="405" alt="image" src="https://github.com/powersync-ja/powersync-supabase-flutter-todolist-demo/assets/46312751/7ad85579-8a45-4a06-a6ae-c5f9018ca521">
